### ISSUE TITLE
no-duplicates should remove duplicate elements

### ISSUE THREAD

**Issue by nickofthyme** [benmosher/eslint-plugin-import#1812]
[`no-duplicates`](https://github.com/benmosher/eslint-plugin-import/blob/v2.20.2/docs/rules/no-duplicates.md) should remove duplicate Imported values.

This case is most commonly seen when resolving merge conflicts where identical imports can be copied. It would be helpful if the fix were able to de-duplicate the imported values themselves not only the paths.

```ts
// bad
import { foo } from '../foo'
import { bar } from '../bar'
import { foo } from '../foo'

// current fix
import { foo, foo } from '../foo' // <-- creates duplicate imported values
import { bar } from '../bar'

// idealized fix
import { foo } from '../foo' // <-- removed duplicated value
import { bar } from '../bar'
```

### Edge cases

```ts
// used with `as`
import { foo as foo1 } from '../foo'
import { bar } from '../bar'
import { foo as foo2 } from '../foo'

// idealized fix to avoid naming collisions 
import { foo as foo1, foo as foo2 } from '../foo' // <-- removed duplicated value
import { bar } from '../bar'
```

**Comment by ljharb** [comment]
The `import { foo, foo }` case imo is a bug. You didn't include "current fix" info with the other cases so off the top of my head I'm not sure if those are a bug or an enhancement.

Either way, happy to review a PR making these changes.

**Comment by nickofthyme** [comment]
Sorry, updated the description to reflect current fixes for edge cases. It looks like the second edge case produces the expected fix.

Ok I'll put up a PR when I have the time and ping you. Thanks

**Comment by ljharb** [comment]
@fingerartur nice! since nick never sent a PR, if you could add tests (that's very important) and send a PR, that would be most appreciated.
